crypto/store/store_result.c

```diff
@@ -153,8 +153,17 @@ int ossl_store_handle_load_result(const OSSL_PARAM params[], void *arg)
         goto err;
     ERR_pop_to_mark();
 
-    if (*v == NULL)
-        ERR_raise(ERR_LIB_OSSL_STORE, ERR_R_UNSUPPORTED);
+    if (*v == NULL) {
+        const char *prov_name = NULL, *hint = "";
+
+        if (provider != NULL)
+            prov_name = OSSL_PROVIDER_get0_name(provider);
+        if (prov_name == NULL)
+            prov_name = "(none)";
+        if (!OSSL_PROVIDER_available(libctx, "default"))
+            hint = ":maybe need to load the default provider?";
+        ERR_raise_data(ERR_LIB_OSSL_STORE, ERR_R_UNSUPPORTED, "provider=%s%s", prov_name, hint);
+    }
```

Comment:
Again, I'd suggest a simple rewrite here, where absent a provider, instead if "(none)", suppress the entire "provider=" part, and if the default provider is loaded (no hint), perhaps suppress the entire detail string.

Reply:
done